canopen/objectdictionary/__init__.py

```diff
@@ -116,13 +116,14 @@ class Record(MutableMapping):
     #: Description for the whole record
     description = ""
 
-    def __init__(self, name, index):
+    def __init__(self, name, index, StorageLocation):
         #: The :class:`~canopen.ObjectDictionary` owning the record.
         self.parent = None
         #: 16-bit address of the record
         self.index = index
         #: Name of record
         self.name = name
+        self.StorageLocation = StorageLocation
```

Comment:
Use a default value of None here so it can be easily identified that it was not present in the EDS.

Add documentation about the attribute by using ```#: Some text```.